### PR TITLE
Add warning message when post is rejected

### LINKED ISSUE #21: Reject Posts option should give feedback explaining that it is related to the post content
Reject Posts give no feedback as to why the post was rejected. It is indistinguishable from some type of connectivity issue. 

When the plugin rejects a post because the feature is enabled, it should show an ephemeral post explaining that the content of the post was rejected because it contains unacceptable terms.

### PR DESCRIPTION
**Pull request by nathanaelhoun**
#### Summary

- [x] Send an ephemeral post to the user when his post is rejected,
- [x] The content of the warning post is configurable in `config.json`,
- [x] `README.md` updated with gifs.

image

#### Ticket Link

Fixes #21

### CODE REVIEW

File: server/plugin.go [hunk 1]
```diff
@@ -39,6 +40,10 @@ func (p *Plugin) FilterPost(post *model.Post) (*model.Post, string) {
 	for i, word := range words {
 		if p.WordIsBad(word) {
 			if configuration.RejectPosts {
+				p.API.SendEphemeralPost(post.UserId, &model.Post{
+					ChannelId: post.ChannelId,
+					Message:   fmt.Sprintf(configuration.WarningMessage, word),
```
Comment (hanzei) [inline server/plugin.go:45]: What happens if a users tries to post a bad word as response to a thread? Does the ephemeral post get posted in the RHS?

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for jumping on this :+1:

File: server/plugin.go (hunk 1 above)
Comment (nathanaelhoun) [reply server/plugin.go:45]: Just tested this: the ephemeral post get posted in the RHS. Is that a problem? I don't know if we can post an ephemeral message in the RHS (tested both with `post.RootId` and `post.ParentId` and it doesn't work)

Comment (hanzei) [reply server/plugin.go:45]: Yes, you can post an ephemeral message in the RHS by setting `RootId`. `post.RootId` already contains the thread `RootId`, which you can use.

Comment (nathanaelhoun) [reply server/plugin.go:45]: Thanks for the input!

File: plugin.json [hunk 2]
```diff
@@ -37,4 +45,4 @@
     "header": "",
     "footer": ""
   }
-}
+}
```
Comment (hanzei) [inline plugin.json:48]: Nit pick: Why remove the newline?

**Review by hanzei — APPROVED** [review]
:tada:

File: plugin.json (hunk 2 above)
Comment (nathanaelhoun) [reply plugin.json:48]: It is my auto-formatter

File: README.md [hunk 3]
```diff
@@ -24,4 +23,10 @@ This plugin allows you to censor profanity on your Mattermost server. The plugin
 
 ### Usage
 
-You can edit the bad words list in **System Console > Plugins > Profanity Filter > Bad Words list**.
+You can edit the bad words list in **System Console > Plugins > Profanity Filter > Bad words list**.
+
+Choose to either censor the bad words with a character or reject the post with a custom warning message:
+
+![Post rejected by the plugin](./.github/post-rejected.gif)
```
Comment (hanzei) [inline README.md:30]: `.github` is usually used for config files. Other plugins use an `images` directory. Mind moving the files?

Comment (nathanaelhoun) [reply README.md]: Of course!

File: plugin.json (hunk 2 above)
Comment (hanzei) [reply plugin.json:48]: Would you mind manually reverting your change?

Comment (nathanaelhoun) [reply plugin.json]: Not at all :-)

**Comment by hanzei** [comment]
@nathanaelhoun Heads up that there is a merge conflict

**Comment by nathanaelhoun** [comment]
@hanzei Thanks for the heads up! This has been fixed :-)

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Rejected post UI is clear
- Tested on desktop and mobile
- No issues found

LGTM!

**Review by larkox — APPROVED** [review]
LGTM

**Review by jfrerich — APPROVED** [review]

**Comment by mickmister** [comment]
It looks like in the `post-rejected` gif that the post is still posted, above the ephemeral message. Is this post supposed to be removed from the user's UI after being rejected?

**Comment by nathanaelhoun** [comment]
@mickmister You are right, I shouldn't have cut the right side of the capture. Should I submit a PR to replace with this one?
ezgif com-optimize(2)

**Comment by mickmister** [comment]
@nathanaelhoun It would seem that webapp itself displays the `Retry - Cancel` appears when a given post fails for any reason? This functionality seems fine to me.

Up to you whether you want to create a PR for the gif change :)

**Comment by nathanaelhoun** [comment]
@mickmister Exactly, the webapp itself provide the `Retry - Cancel` button, for a post blocked for any reason (at least for network issue, db issue, or rejected by a plugin)

I know best is the enemy of good, but I want to make the work correctly: I have submitted #29 :D

**Pull request merged**